app/models/model.rb

```diff
@@ -4,6 +4,13 @@ class Model < ApplicationRecord
   extend FriendlyId
   friendly_id :name
   include Parameterizable::ByIdName
+  include ::Foreman::HookSubscribers::Observable
+
+  set_hook :created, on: :create
+  set_hook :updated, on: :update
+  set_hook :destroyed, on: :destroy do |model|
+    { id: model.id, name: model.name }
+  end
 
   before_destroy EnsureNotUsedBy.new(:hosts)
   has_many_hosts
```

Comment:
One think is missing tho - we would like to have subsciption API for AR models as well as regular Ruby objects. The Observable class only works with AR (`after_commit`), please either refactor it or create a different Observable class for regular Ruby objects. Basically, I'd expect that it would define some "triggering" methods which could be called programmatically. Also add a test for it.

This is where I think `model/action` naming convention is weird, I think a hook should be simply a string that's ensured to be just `[a-z0-9_]` for example `model_created` so implementations can take advantage of it - e.g. "shell" plugin would expect a filename named like this.